pkg/repository/testresult/filter.go

```diff
@@ -6,128 +6,128 @@ import (
 	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
 )
 
-type filter struct {
-	name       string
-	lastNDays  int
-	startDate  *time.Time
-	endDate    *time.Time
-	statuses   testkube.TestSuiteExecutionStatuses
-	page       int
-	pageSize   int
-	textSearch string
-	selector   string
-}
-
-func NewExecutionsFilter() *filter {
-	result := filter{page: 0, pageSize: PageDefaultLimit}
+type FilterImpl struct {
```

Comment:
Looks like some dotnet or Java 

Do we have any convention for naming? In go there is interface and concrete implementation like repository and mongo repository or sql repository

Reply:
The issue is that Interface type cannot be serialized over wire so I must pass the concrete implementation